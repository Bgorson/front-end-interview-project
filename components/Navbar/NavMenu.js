import {useState} from 'react';
import styled from 'styled-components';
import {StyledLink} from './Navbar';
import HamburgerMenu from 'react-hamburger-menu';

const StyledMobileMenu = styled.ul `{  
        list-style-type: none;
        display:flex;
        position:absolute;
        top:4rem;
        z-index:1000;
        background:#222;
        flex-direction: column;
        border-radius: 3px;
}`;
const StyledDesktopMenu = styled.ul `{  
        list-style-type: none;
        display:flex;
        @media (max-width: 600px) {
                display: none;
            }
}`;

const StyledItems= styled.li `{
    font-size: 20px;
    margin:1rem;
    list-style-type: none;
}`;

export default function NavMenu() {
    const [open, setOpen]= useState(false);
    return (
        <>
            <HamburgerMenu
                className={'hamburger'}
                isOpen={open}
                menuClicked={()=>setOpen(!open)}
                width={18}
                height={15}
                strokeWidth={1}
                rotate={0}
                color='white'
                borderRadius={0}
                animationDuration={0.5}
            />
            {open?
                <StyledMobileMenu>
                    <StyledItems><StyledLink href="#">Stores</StyledLink></StyledItems>
                    <StyledItems><StyledLink href="#">Contact Us</StyledLink></StyledItems>
                </StyledMobileMenu>:null}
            <StyledDesktopMenu>
                <StyledItems><StyledLink href="#">Stores </StyledLink></StyledItems>
                <StyledItems><StyledLink href="#">Contact Us</StyledLink></StyledItems>
            </StyledDesktopMenu>
        </>
    );}