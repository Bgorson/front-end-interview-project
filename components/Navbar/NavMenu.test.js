
import renderer from 'react-test-renderer';
import NavMenu from './NavMenu';

it('should render correctly', () => {
    const component = renderer.create(<NavMenu/>).toJSON();

    expect(component).toMatchSnapshot();
});