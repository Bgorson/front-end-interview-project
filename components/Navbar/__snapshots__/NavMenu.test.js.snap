// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
Array [
  <div
    className="hamburger"
    onClick={[Function]}
    style={
      Object {
        "height": "15px",
        "position": "relative",
        "transform": "rotate(0deg)",
        "width": "18px",
      }
    }
  >
    <span
      style={
        Object {
          "background": "white",
          "borderRadius": "0px",
          "display": "block",
          "height": "1px",
          "marginTop": "-0.5px",
          "position": "absolute",
          "transform": "translate3d(0,0,0) rotate(0)",
          "transformOrigin": "center",
          "transitionDuration": "0.5s",
          "transitionTimingFunction": "ease",
          "width": "100%",
        }
      }
    />
    <span
      style={
        Object {
          "background": "white",
          "borderRadius": "0px",
          "display": "block",
          "height": "1px",
          "marginTop": "-0.5px",
          "opacity": "1",
          "position": "absolute",
          "top": "7.5px",
          "transformOrigin": "center",
          "transitionDuration": "0.125s",
          "transitionTimingFunction": "ease-out",
          "width": "100%",
        }
      }
    />
    <span
      style={
        Object {
          "background": "white",
          "borderRadius": "0px",
          "display": "block",
          "height": "1px",
          "marginTop": "-0.5px",
          "position": "absolute",
          "transform": "translate3d(0,15px,0) rotate(0)",
          "transformOrigin": "center",
          "transitionDuration": "0.5s",
          "transitionTimingFunction": "ease",
          "width": "100%",
        }
      }
    />
  </div>,
  <ul
    className="NavMenu__StyledDesktopMenu-sc-194gy50-1 iyvdhz"
  >
    <li
      className="NavMenu__StyledItems-sc-194gy50-2 gRmCyW"
    >
      <a
        className="Navbar__StyledLink-sc-1h1ckcl-2 cvqca-d"
        href="#"
      >
        Stores 
      </a>
    </li>
    <li
      className="NavMenu__StyledItems-sc-194gy50-2 gRmCyW"
    >
      <a
        className="Navbar__StyledLink-sc-1h1ckcl-2 cvqca-d"
        href="#"
      >
        Contact Us
      </a>
    </li>
  </ul>,
]
`;
